kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -300,6 +356,115 @@ object Abort:
       */
     inline def recover[E]: RecoverOps[E] = RecoverOps(())
 
+    final case class FoldOps[E](dummy: Unit) extends AnyVal:
```

Comment:
Could you make these methods resemble their counterparts in `Result`? It'd provide a more consistent experience. Feel free to make improvements to `Result` if needed as well.

Reply:
It seems we could avoid code duplication by introducing `runWith` taking a continuation function and then using the regular `Result` APIs to transform it

Reply:
I've added `runWith` and am using it for `run`, `runPartial`, and `fold`. Shall I update other methods to use it as well? Is there a performance concern with building other `Abort` handlers on top of `runWith`?

Reply:
There shouldn't be a significant performance penalty. Yes, feel free to refactor other methods! :) Do you think we should keep `runWith` public? If not, we could make it `private` and `inline` to reduce potential overhead. We can't inline directly in user code, though. Effect handling generates large trees.

Reply:
My attempt to make `Abort` and `Result` more consistent has led to a pretty massive change. Since I want to overload `fold` in `Abort`, this means un-currying and changing the parameter order in `Result`. Let me know if you have an issue with this.

Reply:
Sounds good. My concern is more uniformity between the APIs so the user can form a mental model that works for both